### PR TITLE
Improved status_isdead and status_get_status_data

### PR DESCRIPTION
**Pull request by AoShinRO**
* **Addressed Issue(s)**: 

* **Server Mode**: 

* **Description of Pull Request**:  i have some issues  refering nullptr at status_get_status_data(bl) called by status_isdead

### CODE REVIEW

File: src/map/status.cpp [hunk 1]
```diff
@@ -9223,7 +9223,7 @@ std::vector<e_race2> status_get_race2(struct block_list *bl)
  */
 int status_isdead(struct block_list *bl)
```
Comment (Atemo) [inline src/map/status.cpp:9224]: Can we use ref and return a bool instead?

Comment (AoShinRO) [reply src/map/status.cpp]: it's that?

Comment (AoShinRO) [reply src/map/status.cpp]: if change status_getdata  to ref is bad idea, say and i revert it  @Atemo

**Comment by AoShinRO** [comment]
@Lemongrass3110 it work's

**Pull request merged**